homeassistant/components/generic/config_flow.py

```diff
@@ -314,19 +330,49 @@ async def async_step_user(
                         # The automatically generated still image that stream generates
                         # is always jpeg
                         user_input[CONF_CONTENT_TYPE] = "image/jpeg"
+                    self.user_input = user_input
+                    self.title = name
 
-                    return self.async_create_entry(
-                        title=name, data={}, options=user_input
+                    # temporary preview for user to check the image
+                    self.context["preview_cam"] = GenericCamera(
```

Comment:
I'm not sure it's safe to put non serializable data in the context. I think the config flow http server endpoints may try to serialize the flow result in progress. I'd just put the user input in the context and create the camera instance when needed.

Reply:
Ok, done 7f13bbb. Makes no major difference either way so ok by me.